src/syntax-tree/ast.ts

```diff
@@ -1311,11 +1311,7 @@ export class VarAssignmentStmt extends Statement implements VariableContainer {
     }
 
     onInsertInto(insertCode: Expression) {
-        if (insertCode instanceof ListLiteralExpression) {
-            this.dataType = TypeChecker.getElementTypeFromListType(insertCode.returns);
-        } else {
-            this.dataType = insertCode.returns;
-        }
+        this.dataType = insertCode.returns; //#344
```

Comment:
Not really sure why it was the way it was before. It really seems like this was a remnant of some old type checking. I did not see this breaking anything once I removed it.